ament_package/package.py

```diff
@@ -98,9 +98,21 @@ def validate(self):
 
         if not self.name:
             errors.append('Package name must not be empty')
-        # only allow lower case alphanummeric characters and underscores
+        # assume packages are catkin unless a build_type is exported.
+        build_type = 'catkin'
+        build_types = [ex.contents for ex in self.exports if ex.tagname == 'build_type']
+        if len(build_types) > 1:
+            # this is bad, but first I must ask if we should make this an error.
```

Comment:
Sounds sensible to me. The rep prevents it it makes sense to raise an error for it. (Then again I'm not expert in either ament or catkin so it's just my outsider's opinion)